src/snapshot/set.rs

```diff
@@ -67,7 +74,32 @@ impl Plugin for SnapshotSetPlugin {
             SaveWorld,
             (SaveWorldSet::Checksum, SaveWorldSet::Snapshot).chain(),
         )
+        .configure_sets(
+            AdvanceWorld,
+            (
+                AdvanceWorldSet::First,
+                AdvanceWorldSet::Main,
+                AdvanceWorldSet::Last,
+            )
+                .chain(),
+        )
         .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::EntityFlush))
-        .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::DataFlush));
+        .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::DataFlush))
+        .add_systems(
+            AdvanceWorld,
+            apply_deferred
+                .after(AdvanceWorldSet::First)
+                .before(AdvanceWorldSet::Main),
+        )
+        .add_systems(
+            AdvanceWorld,
+            apply_deferred
+                .after(AdvanceWorldSet::Main)
+                .before(AdvanceWorldSet::Last),
+        )
+        .add_systems(
+            AdvanceWorld,
+            (|world: &mut World| world.run_schedule(GgrsSchedule)).in_set(AdvanceWorldSet::Main),
```

Comment:
Since `AdvanceWorld` is linear and mostly just runs `GgrsSchedule`, and not really intended for the user to modify (?) I think its executor should be set to single-threaded. The user could always change it back if they do something weird.

The user can change this, but I think it's best if we provide the most sensible defaults.